### PR TITLE
Fix error by properly checking zero for floats

### LINKED ISSUE #309: 10431 Error – Item Amount Invalid at Checkout
I came across a bug when using the live environment, **and** ~~which wasn't manifesting itself using~~ the sandbox... I am seeing a [10431 error](https://developer.paypal.com/docs/classic/api/errors/#10431) on the checkout page.
It is also possibly related to using a coupon code since in my limited testing it occurred after adding a coupon.
I also don't get the error if I click the 'Check out with PayPal' button on the cart page, coupon code or not.

**Environment**
- WordPress `V4.8`
- WooCommerce `V3.1.1`
- WooCommerce PayPal Express Checkout Gateway `V1.4.2`

**Steps to Reproduce**
- Add an item worth $24.95 to the cart
- Apply a coupon code which gives a $4.95 discount
- Click the 'Proceed to checkout' button 
- Select PayPal Express payment option
- Click the 'Continue to payment' button
- If I remove the coupon, it works as expected

It is also reported over on the [WordPress Plugin page](https://wordpress.org/support/topic/paypal-error-10431/).

**Screenshot**
screen shot 2017-08-01 at 9 54 05 am

### PR DESCRIPTION
**Pull request by bor0**
Fixes #309.

### CODE REVIEW

**Comment by hamoderr** [comment]
I know I was not invited to review. My acceptance is just an indication that that making these changes locally has resolved the issue for me.

**Comment by bor0** [comment]
Your local testings imply your invitation for review. Glad it's resolved your issue.

Thanks!

File: includes/class-wc-gateway-ppec-client.php [hunk 1]
```diff
@@ -547,7 +547,7 @@ protected function get_details( $details, $discounts, $rounded_total, $total ) {
 				// Add line item to make up different between WooCommerce
 				// calculations and PayPal calculations.
 				$diff = round( $details['total_item_amount'] - $rounded_total, $decimals );
-				if ( $diff != 0 ) {
+				if ( abs( $diff ) < 0.0001 ) {
```
Comment (gedex) [inline includes/class-wc-gateway-ppec-client.php]: what's the reasoning to choose the number `0.0001` ? Is it mentioned in PPEC doc?

Comment (bor0) [inline includes/class-wc-gateway-ppec-client.php]: It uses the formula `|a - b| < e` to see how close a and b are for a small e (the smaller the better). In this case b = 0 and e = 0.0001.

As far as I checked debug logs we were sending PP a very small number which was caused by the calculation above on floats, so PP didn't like that number.

**Pull request merged**